packages/cloud-functions/src/cico/rampWebhook.ts

```diff
@@ -0,0 +1,50 @@
+import crypto from 'crypto'
+import stableStringify from 'fast-json-stable-stringify'
+import * as functions from 'firebase-functions'
+import { readFileSync } from 'fs'
+import { RAMP_KEY } from '../config'
+import { saveTxHashProvider } from '../firebase'
+import { Provider } from './Provider'
+
+const rampKey = readFileSync(`./config/${RAMP_KEY}`).toString()
+
+function verifyRampSignature(signature: string | undefined, body: any) {
+  if (!signature || !body) {
+    return false
+  }
+
+  const verifier = crypto.createVerify('sha256')
+  verifier.update(Buffer.from(stableStringify(body)))
+  return verifier.verify(rampKey, signature, 'base64')
+}
+
+const RAMP_SIGNATURE_HEADER = 'X-Body-Signature'
+const RAMP_RELEASED = 'RELEASED'
+const RAMP_RELEASING = 'RELEASING'
+
+export const rampWebhook = functions.https.onRequest((request, response) => {
+  if (verifyRampSignature(request.header(RAMP_SIGNATURE_HEADER), request.body)) {
+    const { type, purchase } = request.body
```

Comment:
Can you please create a Ramp request body type so that someone who's new to the code knows what to expect?

Reply:
Doing this helped me realize that they changed their API! O.O The tx hash now comes in a different place.